src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1418,6 +1451,45 @@ MonodroidRuntime::typemap_managed_to_java (MonoReflectionType *type, const uint8
 	return embeddedAssemblies.typemap_managed_to_java (type, mvid);
 }
 
+#if defined (WINDOWS)
+const char*
+MonodroidRuntime::get_my_location ()
+{
+	HMODULE hm = NULL;
+
+	if (GetModuleHandleEx (GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
+	                      GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+	                      (LPCSTR) &get_xamarin_android_msbuild_path, &hm) == 0) {
+		int ret = GetLastError ();
+		log_warn (LOG_DEFAULT, "GetModuleHandle failed, error = %d\n", ret);
+		return nullptr;
+	}
+
+	WCHAR path[MAX_PATH * 2];
+	if (GetModuleFileNameW (hm, path, sizeof(path)) == 0) {
+		int ret = GetLastError ();
+		log_warn (LOG_DEFAULT, "GetModuleFileName failed, error = %d\n", ret);
```

Comment:
Given that we're immediately returning, there is no "context" to this message. *Anything* could cause it.  Additionally, this message looks very similar to the `GetModuleFileName failed` message, meaning *I* will get confused when trying to deduce what went wrong. :-)

The messages should thus be updated to mention what's happening, e.g.

```cpp
log_warn (LOG_DEFAULT, "Unable to get filename to `libmono-android.debug.dll`; GetModuleFileName returned %d\n", ret);
```